### PR TITLE
feat: Add Azure blob support

### PR DESCRIPTION
**Pull request by vasspilka**
# Summary

We are adding support for Azure blob by creating a new service. https://github.com/ash-project/ash_storage/issues/14

The implementation is tested with a local Azurite instance (started via docker) note that if you don't have the image the test will download it during setup. This implementation has been also tested in a real Ash application using `ash_storage` with a staging Azure blob instance. 

I've made attempts to think through any edge cases that one might encounter and they should be covered. However I'm not an expert with Azure blob so I might have missed something. The core happy paths work well and without issues.

Finally want to note that the changes to the doc `documentation/topics/direct-uploads.md` around headers were done because Azure requires extra request headers. This makes the docs more consistent even if it's not mandatory for S3 (everything still works when passing the headers)

# Contributor checklist

Leave anything that you believe does not apply unchecked.

- [x] I accept the [AI Policy](https://github.com/ash-project/.github/blob/main/AI_POLICY.md), or AI was not used in the creation of this PR.
- [ ] Bug fixes include regression tests
- [ ] Chores
- [x] Documentation changes
- [x] Features include unit/acceptance tests
- [ ] Refactoring
- [ ] Update dependencies

### CODE REVIEW

**Comment by zachdaniel** [comment]
Looks generally good to me 👍 LMK when ready for a full review.

**Comment by vasspilka** [comment]
Hey @zachdaniel I just tested this branch with a real Azure instance, and did an additional self-review. Should be good to go. Let me know if we should do any adjustments.

**Review by zachdaniel — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,3 +5,10 @@
 ### Added
 
 - Initial project setup
+- Added `AshStorage.Service.AzureBlob` for Azure Blob Storage uploads, downloads, URLs, and direct uploads.
+- Added Azurite-backed live integration coverage for `AshStorage.Service.AzureBlob`.
+- Documented Azure SAS/CORS requirements and avoided persisting literal Azure credentials on blob records.
+
+### Changed
```
Comment (zachdaniel) [inline CHANGELOG.md]: no need to manually modify the changelog. We update it from commit messages 😄

**Comment by vasspilka** [comment]
@zachdaniel All done 💜

**Comment by zachdaniel** [comment]
🚀 Thank you for your contribution! 🚀
